IBAnimatable/IBAnimatableTests/TestProtocols/PaddingDesignableTests.swift

```diff
@@ -15,37 +15,126 @@ protocol PaddingDesignableTests: class {
-    XCTAssertEqual(element.rightViewMode, .always)
+  func _testRightTextPadding() {
+    testTextPadding(edge: .right(21))
   }
 
-  func _testPaddingSide() {
-    element.paddingSide = 20.0
-    let paddingView = UIView(frame: CGRect(x: 0, y: 0, width: 20.0, height: 1))
-    XCTAssertEqual(element.rightView?.frame, paddingView.frame)
-    XCTAssertEqual(element.rightViewMode, .always)
-    XCTAssertEqual(element.leftView?.frame, paddingView.frame)
-    XCTAssertEqual(element.leftViewMode, .always)
+  func _testSideTextPadding() {
+    testTextPadding(edge: .sides(22))
+  }
+
+  func _testLeftEditPadding() {
+    testEditPadding(edge: .left(23))
+  }
+
+  func _testRightEditPadding() {
+    testEditPadding(edge: .right(24))
+  }
+
+  func _testSideEditPadding() {
+    testEditPadding(edge: .sides(25))
+  }
+
+  func _testLeftPlaceholderPadding() {
+    testPlaceholderPadding(edge: .left(26))
+  }
+
+  func _testRightPlaceholderPadding() {
+    testPlaceholderPadding(edge: .right(27))
+  }
+
+  func _testSidePlaceholderPadding() {
+    testPlaceholderPadding(edge: .sides(28))
+  }
+
+  // MARK: - Test Helper Methods
+
+  private func testTextPadding(edge: Edge) {
+
+    let originalRect = element.textRect(forBounds: element.bounds)
+    let mockRect = UIEdgeInsetsInsetRect(originalRect, edge.insets)
+
+    switch edge {
+    case .left(let padding): element.leftTextPadding = padding
+    case .right(let padding): element.rightTextPadding = padding
+    case .sides(let padding): element.sideTextPadding = padding
+    }
+
+    let newRect = element.textRect(forBounds: element.bounds)
+    XCTAssertEqual(newRect, mockRect)
+
+  }
+
+  private func testEditPadding(edge: Edge) {
+
+    let originalRect = element.editingRect(forBounds: element.bounds)
+    let mockRect = UIEdgeInsetsInsetRect(originalRect, edge.insets)
+
+    switch edge {
+    case .left(let padding): element.leftEditPadding = padding
+    case .right(let padding): element.rightEditPadding = padding
+    case .sides(let padding): element.sideEditPadding = padding
+    }
+
+    let newRect = element.editingRect(forBounds: element.bounds)
+    XCTAssertEqual(newRect, mockRect)
+    print(mockRect)
+    print(newRect)
+
+  }
+
+  private func testPlaceholderPadding(edge: Edge) {
+
+    element.placeholder = "placeholderRect won't get called if this is nil"
+    let originalRect = element.placeholderRect(forBounds: element.bounds)
+    let mockRect = UIEdgeInsetsInsetRect(originalRect, edge.insets)
+
+    switch edge {
+    case .left(let padding): element.leftPlaceholderPadding = padding
+    case .right(let padding): element.rightPlaceholderPadding = padding
+    case .sides(let padding): element.sidePlaceholderPadding = padding
+    }
+
+    let newRect = element.placeholderRect(forBounds: element.bounds)
+    XCTAssertEqual(newRect, mockRect)
+
+  }
+
+}
+
+// MARK: - Helper Enum
+
+fileprivate enum Edge {
+  case left(CGFloat)
```

Comment:
I like this idea, can we use it in our real API for the library? But don't know how to support IB yet.